### PR TITLE
Added two new fuctions to dict API

### PR DESCRIPTION
**Pull request by lgg2**
Hello,

I have added two new functions to iterate a dictionary without the use of a callback. I have also include a test purpose file.

Feel free to correct the changes. I think that are usefull.

P.D.: Sorry, I see in the commit that I have include the "test_mod" executable. Delete it.

Thanks.

### CODE REVIEW

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
I'm OK with this in concept, but see my notes.

Also, all APIs need to range check the input arguments - look for NULL pointers, etc.

File: pdfio-dict.c [hunk 1]
```diff
@@ -524,6 +524,36 @@ pdfioDictIterateKeys(
...
+{
+	return (dict->num_pairs);
+}
+
+//
+// 'pdfioDictGetKeyByIndex()' - Get name and type of a key by index from a dictionary
+//
+// Alternative form to enumerate the keys of a dictionary
+
+bool								// O - Value
```
[the remark this answers is not in the document]
Comment (lgg2) [reply pdfio-dict.c:542]: The problem that I see only returning a string with the key It is that also It is of interest get the type with the save call.

**Comment by lgg2** [comment]
I'm not a full time programmer (only for hobby and some times for solve something in the work) and It is also my first "pull request" (also again in a very rudimentary way).

The reason for this two functions It is no need the use of a callback for each time that review for a dictionary keys.

**Comment by michaelrsweet** [comment]
[master afac835] Add pdfioDictGetKey and pdfioDictGetNumPairs APIs (Issue #63)

**Pull request closed without merging**